tools/buildtools/schc.py

```diff
@@ -0,0 +1,923 @@
+
+    # Label to jump into this command
+    labelName: str
+
+    # True if this expression is allowed to change between `_s` and `_l` versions of the same command, like if the original script used a suffix-less version of the command
+    # False if the original script used a command with a `_s`/`_l` suffix
+    canChange: bool
+
+    # The branch target can be either a str (label) or None (no branch target)
+    branchTarget: str|None = None
+
+    def __str__(self) -> str:
+        ret = f"/* {self.index:03} */ {self.labelName:<24}: {self.token.tokenLiteral}"
+        if self.canChange:
+            ret += "*"
+        if self.args is not None:
+            ret += f" ({self.args.tokenLiteral})"
+        if self.branchTarget is not None:
+            ret += f" -> {self.branchTarget}"
+        return ret
+
+# Takes a tree and linearizes it, preserving the control flow by using labels (user-defined or autogenerated)
+def convertTreeIntoLabeledList(tree: list[Expression], index: int = 0) -> tuple[list[LabeledExpression], int]:
+    result: list[LabeledExpression] = []
+    # To track the current label
+    labelName: str|None = None
+
+    for expr in tree:
+        token = expr.token
+        if token.tokenType == TokenType.LABEL:
+            # Keep the labelname but ignore the label itself
+            labelName = token.tokenLiteral
+            continue
+
+        tokenProperties = token.getProperties()
+
+        left = expr.left
+        right = expr.right
+
+        # Some commands need to invert theirs `if_`/`else` bodies to match the corresponding command
+        if tokenProperties.needsToInvert:
+            left, right = right, left
+
+        # Invert the bodies if the expression was negated
+        if expr.negated:
+            left, right = right, left
+
+        currentIndex = index
+        index += 1
+
+        # Linealize the left body
+        subResults, index = convertTreeIntoLabeledList(left, index)
+
+        # Expressions always jump into the right body if their check evaluates to True
+        targetLabel = None
+        if len(right) == 1 and right[0].token.getProperties().isUnconditionalBranch:
+            # If an `if_` only has 1 expression and it is a branch then incorporate it as part of the `if_`,
+            # avoiding redundant branches
+            branchExpr = right[0]
+            if branchExpr.args is None:
+                fatalError(f"Branch command without arguments?", branchExpr.token.filename, branchExpr.token.lineNumber, branchExpr.token.columnNumber)
+            targetLabel = branchExpr.args.tokenLiteral.strip()
+        else:
+            sub, index = convertTreeIntoLabeledList(right, index)
+            subResults += sub
+            if len(sub) != 0:
+                # Get the label name of the first expression
+                targetLabel = sub[0].labelName
+
+        # If there's no user-defined label, then autogenerate one. Use dots to avoid name crashes with user-defined ones
```

Comment:
```suggestion
        # If there's no user-defined label, then autogenerate one. Use dots to avoid name clashes with user-defined ones
```
The japanengrish is getting to you monkaS